lib/dsc-lib/src/functions/cidr_host.rs

```diff
@@ -0,0 +1,329 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use super::Function;
+use crate::configure::context::Context;
+use crate::functions::{FunctionArgKind, FunctionCategory, FunctionMetadata};
+use crate::DscError;
+use ipnetwork::{IpNetwork, Ipv4Network, Ipv6Network};
+use rust_i18n::t;
+use serde_json::Value;
+use tracing::debug;
+
+#[derive(Debug, Default)]
+pub struct CidrHost {}
+
+impl Function for CidrHost {
+    fn get_metadata(&self) -> FunctionMetadata {
+        FunctionMetadata {
+            name: "cidrHost".to_string(),
+            description: t!("functions.cidrHost.description").to_string(),
+            category: vec![FunctionCategory::String],
+            min_args: 2,
+            max_args: 2,
+            accepted_arg_ordered_types: vec![
+                vec![FunctionArgKind::String],
+                vec![FunctionArgKind::Number],
+            ],
+            remaining_arg_accepted_types: None,
+            return_types: vec![FunctionArgKind::String],
+        }
+    }
+
+    fn invoke(&self, args: &[Value], _context: &Context) -> Result<Value, DscError> {
+        debug!("{}", t!("functions.cidrHost.invoked"));
+
+        let cidr_string = args[0].as_str().ok_or_else(|| {
+            DscError::FunctionArg(
+                "cidrHost".to_string(),
+                t!("functions.cidrHost.invalidNetwork").to_string(),
+            )
```

Comment:
Since parameter types are pre-validated, this error will never get returned so no need to have this string stored in the locales.  You should just `as_str().unwrap();` as it should never panic.